frontend/src/libs/components/order-card/order-card.tsx

```diff
@@ -1,70 +1,80 @@
 import { PlainSvgIconName } from '~/libs/enums/enums.js';
-import { getValidClassNames, pluralizeString } from '~/libs/helpers/helpers.js';
+import { getValidClassNames } from '~/libs/helpers/helpers.js';
+import {
+  useAppDispatch,
+  useAppSelector,
+  useEffect,
+} from '~/libs/hooks/hooks.js';
+import { actions as orderActions } from '~/slices/orders/order.js';
+import { selectOrder, selectOrderData } from '~/slices/orders/selectors.js';
 
 import { PlainSvgIcon } from '../plain-svg-icon/plain-svg-icon.js';
-import {
-  type DriverProfileInfo,
-  type OrderCurrentStatus,
-  type OrderedTruckInfo,
-  type OrderInitialStatus,
-} from './libs/types/types.js';
 import styles from './styles.module.scss';
 
 type Properties = {
-  driver: DriverProfileInfo;
-  truck: OrderedTruckInfo;
-  initialStatus: OrderInitialStatus;
-  currentStatus: OrderCurrentStatus;
+  className?: string;
+  isDriverShown?: boolean;
 };
 
 const OrderCard: React.FC<Properties> = ({
-  driver: { firstName, lastName, profileURL },
-  truck: { licensePlate },
-  initialStatus: { startLocation, endLocation },
-  currentStatus: { timespanLastUpdated, location, distanceLeft, timespanLeft },
+  className,
+  isDriverShown = true,
 }: Properties) => {
-  const pluralizedKilometers = pluralizeString('km', distanceLeft);
+  const [order] = useAppSelector(selectOrder);
+  const routeData = useAppSelector(selectOrderData);
+  const startLocation = routeData.origin ?? '';
+  const endLocation = routeData.destination ?? '';
+  const distanceLeft = routeData.distanceAndDuration?.distance.text ?? '';
+  const timespanLeft = routeData.distanceAndDuration?.duration.text ?? '';
+  const licensePlate = order?.shift.truck?.licensePlateNumber ?? '';
+  const price = order?.price;
+  const firstName = order?.shift.driver?.firstName ?? '';
+  const lastName = order?.shift.driver?.lastName ?? '';
+  const profileURL =
+    'https://images.freeimages.com/images/large-previews/962/avatar-man-with-mustages-1632966.jpg?fmt=webp&w=350';
+  const dispatch = useAppDispatch();
```

Comment:
```suggestion
  const [order] = useAppSelector(selectOrder);
  const { origin, destination, distanceAndDuration } = useAppSelector(selectOrderData);
    const { driver, truck } = order?.shift // same for other stuff
  const profileURL =
    'https://images.freeimages.com/images/large-previews/962/avatar-man-with-mustages-1632966.jpg?fmt=webp&w=350'; // Let's not use mocks, if there is no file - show default background.
  const dispatch = useAppDispatch();
```
Not sure why there might be a case when there is no order `order?.` 🤔

Reply:
if user navigates to page by url order/:orderId there is no order in state so it can be undefined during first render